token/services/utils/json/jsession/session.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jsession
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Session local alias for view.Sessions
+type Session = view.Session
+
+type JSession struct {
+	s       Session
+	context context.Context
+}
+
+// New returns a JSON-based session wrapping the passed session.
+func New(context context.Context, s Session) *JSession {
+	span := trace.SpanFromContext(context)
+	span.AddEvent(fmt.Sprintf("Open json session to [%s:%s]", string(s.Info().Caller), s.Info().CallerViewID))
```

Comment:
`s.Info()` is an expensive call (it contains a lock). I would suggest you use it only once or even better you use the `Eval` util:
```
logger.DebugfContext(context, "Open json session: %s", logging.Eval(s.Info))
```